lib/vfscore/pipe.c

```diff
@@ -699,6 +698,9 @@ UK_SYSCALL_R_DEFINE(int, pipe2, int*, pipefd, int, flags)
 {
 	int rc;
 
+	if (flags & O_DIRECT)
+		return -EINVAL;
```

Comment:
Here why are we returning the negative value?

Reply:
If you mean why negative, it's because we're in the context of a raw syscall, which by API convention returns negative error values.

If you mean why error out on this flag, it's because an `O_DIRECT` pipe behaves differently to a regular pipe on Linux, operating in "packet mode" -- essentally like a datagram socket where writes match up with reads 1-to-1. Unikraft sockets don't implement such a mode, and for applications that rely on this behavior it's better to error out early than risk weird bugs.